share/spack/gitlab/cloud_pipelines/stacks/e4s-oneapi/spack.yaml

```diff
@@ -87,6 +91,10 @@ spack:
       variants: threads=openmp
     python:
       version: [3.8.13]
+    ruby:
+      compiler: [gcc] # https://github.com/spack/spack/issues/31954
+    rust:
+      compiler: [gcc] # undefined reference becauase of -nodefaultlibs
```

Comment:
Just a note that, since #32260 was merged, you can:
```yaml
ruby:
  require: "%gcc"
```
to enforce this constraint (i.e. this is not a preference, but something the concretizer need to respect).

Reply:
> to enforce this constraint (i.e. this is not a preference, but something the concretizer need to respect).

I will try it. ```compiler:``` was still compiling some packages with oneapi. Do you expect it to be merged soon? Or should I cherry pick it into my branch?

Reply:
Ugh, forgot one detail. You need to wait for #32260 to be able to specify also:
```
packages:
  all:
    require: "%oneapi"
```
or whatever constraint you want to impose by default.

Reply:
That will probably be merged today or tomorrow